### PR TITLE
Add Advanced Settings switch to Editor Settings

### LINKED ISSUE #30866: Advanced Settings option
**Godot version:**
94aabf5 

**Issue description:**
Recurring problem with some feature proposals is that they **absolutely can't** add any Editor Settings, _because settings are bloated_ and editor becomes complex etc. This is problematic, because more options is sometimes useful. Well, the settings really have some not-so-useful options (probably) and aren't well-organized, but that's for another issue.

Discussion in #30816 spawned an idea for a nice feature that could do something about the problem of editor becoming "too complex" - Advanced Settings option. Basically, upon opening Editor Settings, you'd see only a set of very common options (like editor language or theme) and most of the settings would be hidden (like interface dialogs or file server). To show them, there would be an "Advanced Settings" checkbox.

This way, when you open Editor Settings, you will see stuff that is mostly relevant and you are likely to change. It would keep users from being lost and if someone wants to tweak more things and have option for literally everything, it would be possible. The biggest problem here is determining which options should be considered "advanced".

Also, here's how it looks in VLC, which does it really well:
Basic settings
image

Advanced settings
image

Anyone disagrees? Any proposals for basic and advanced settings?

### PR DESCRIPTION
**Pull request by KoBeWi**
Closes #30866

https://github.com/user-attachments/assets/66909d8d-1d27-41dd-8c0d-855959c49766

Feel free to suggest changes to this list.

Other than that:
- `restore_scenes_on_load` is no longer marked as requiring restart
- enabled code that makes default value settings not stored

### CODE REVIEW

**Comment by Maran23** [comment]
Some suggestions:
[Editor]
- Language
- Dock Tab Style
- Font Size
- Display Scale
- Single Window Mode

[Theme]
- Follow System Theme
- Preset
- Spacing Preset

[Connection]
- Engine Version Update Mode
- Network Mode

[Import]
- Blender Path

[Complection]
- Idle Parse Delay
- Code Complete Delay
- Add String Name Literals
- Add Node Path Literals

[Window Placement] (also in case Godot does allow to run the game in the editor)

[Linuxbsd]
- Prefer Wayland

[Project Manager] (the three settings)

[Android]
- Android SDK Path
- Java SDK Path

[macOS]
- rcodesign

[Windows]
- rcedit
- signtool

File: editor/editor_settings_dialog.cpp [hunk 1]
```diff
@@ -845,8 +850,16 @@ EditorSettingsDialog::EditorSettingsDialog() {
 	search_box->set_h_size_flags(Control::SIZE_EXPAND_FILL);
 	hbc->add_child(search_box);
 
+	advanced_switch = memnew(CheckButton(TTR("Advanced Settings")));
+	hbc->add_child(advanced_switch);
+
+	bool use_advanced = EditorSettings::get_singleton()->get_project_metadata("editor_settings", "advanced_mode", false);
```
[the remark this answers is not in the document]
Comment (AThousandShips) [reply editor/editor_settings_dialog.cpp:856]: I'd say it makes sense per project, depending on what you're doing I'd say it influences what you actually want

[the remark this answers is not in the document]
Comment (KoBeWi) [reply editor/editor_settings_dialog.cpp:856]: I already changed it to store globally.

**Comment by Mickeon** [comment]
Are you by chance implementing this button partially because of https://github.com/godotengine/godot/pull/69012#pullrequestreview-2244280828 ?

**Comment by KoBeWi** [comment]
No, but I can look into integrating that. Maybe after both PRs are merged.

File: editor/editor_settings.cpp [hunk 2]
```diff
@@ -868,32 +879,36 @@ void EditorSettings::_load_defaults(Ref<ConfigFile> p_extra_config) {
...
-	EDITOR_SETTING(Variant::STRING, PROPERTY_HINT_ENUM, "project_manager/default_renderer", "gl_compatibility", "forward_plus,mobile,gl_compatibility")
+	EDITOR_SETTING_BASIC(Variant::STRING, PROPERTY_HINT_ENUM, "project_manager/default_renderer", "gl_compatibility", "forward_plus,mobile,gl_compatibility")
 #elif defined(ANDROID_ENABLED)
 	// Use more suitable rendering method by default.
-	EDITOR_SETTING(Variant::STRING, PROPERTY_HINT_ENUM, "project_manager/default_renderer", "mobile", "forward_plus,mobile,gl_compatibility")
+	EDITOR_SETTING_BASIC(Variant::STRING, PROPERTY_HINT_ENUM, "project_manager/default_renderer", "mobile", "forward_plus,mobile,gl_compatibility")
 #else
-	EDITOR_SETTING(Variant::STRING, PROPERTY_HINT_ENUM, "project_manager/default_renderer", "forward_plus", "forward_plus,mobile,gl_compatibility")
+	EDITOR_SETTING_BASIC(Variant::STRING, PROPERTY_HINT_ENUM, "project_manager/default_renderer", "forward_plus", "forward_plus,mobile,gl_compatibility")
 #endif
```
Comment (m4gr3d) [inline editor/editor_settings.cpp:906]: nit: I know this existed prior to this PR, but this section could be cleaned up by storing the default in a variable and only having that value changes based on the platform it's running on.

File: platform/android/export/export.cpp [hunk 3]
```diff
@@ -42,16 +42,16 @@ void register_android_exporter_types() {
 }
 
 void register_android_exporter() {
-	EDITOR_DEF("export/android/debug_keystore", EditorPaths::get_singleton()->get_debug_keystore_path());
+	EDITOR_DEF_BASIC("export/android/debug_keystore", EditorPaths::get_singleton()->get_debug_keystore_path(), false, true);
```
Comment (m4gr3d) [inline platform/android/export/export.cpp:45]: `false, true` don't seem needed here.

**Review by m4gr3d — CHANGES_REQUESTED** [review]

**Review by m4gr3d — APPROVED** [review]

File: editor/editor_settings_dialog.cpp [hunk 4]
```diff
@@ -845,8 +850,16 @@ EditorSettingsDialog::EditorSettingsDialog() {
 	search_box->set_h_size_flags(Control::SIZE_EXPAND_FILL);
 	hbc->add_child(search_box);
 
+	advanced_switch = memnew(CheckButton(TTR("Advanced Settings")));
```
Comment (m4gr3d) [inline editor/editor_settings_dialog.cpp:853]: Seems there are compile issues with `CheckButton`

**Review by m4gr3d — CHANGES_REQUESTED** [review]

**Review by m4gr3d — APPROVED** [review]

**Comment by akien-mga** [comment]
Needs rebase, then looks good to merge IMO.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**